### PR TITLE
chore: add reporters

### PR DESCRIPTION
**Pull request by KFlash**
There exist two reporters

`aladdin` - gives a detailed info with an arrow marker

`listing` - gives an ESLint 'look a like' output

### CODE REVIEW

**Comment by KFlash** [comment]
@aladdin-add I'm not a designer :) Can you adjust and re-design the reporters and then merge?

In the error callback you pass the 4args like this `listing(a,b,c,d,e, sourceCode)`. 

The `sourceCode` is the code that got parsed.

**Review by aladdin-add — APPROVED** [review]
also need some tests. :)

**Comment by KFlash** [comment]
@aladdin-add How can I test this? And you re-design and merge?

File: src/reporter/listing.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  start: number,
+  end: number,
+  source: string
+) {
+  const lnSpacing = '_'.repeat(90);
+  process.stdout.write(`${lnSpacing}\n\n`);
+  process.stdout.write(
+    `${ColorCodes.Cyan + start}:${end + ColorCodes.Reset}${diagnosticColor1(kind)}${
+      ColorCodes.White + message + ColorCodes.Reset
+    }\n`
```
Comment (KFlash) [inline src/reporter/listing.ts:28]: @aladdin-add  To be equal with ESLint need to count the amount of errors, warning and lint failues so we can print 
'99 errors. 2 warnings. 1 lint issue (3 auto-fixed)'.

**Comment by aladdin-add** [comment]
I will try to introduce jest/mocha, so you can easily test this one(and other similar things). thoughts?

**Comment by KFlash** [comment]
I think mocha already exist in package.json? Anyway. Can you re-design / adjust the layout and merge?

**Pull request merged**